### DISCUSSION TITLE
Search bar issues and inconsistencies

### DISCUSSION THREAD

**Discussion by Squaduck** [ppy/osu#19544] [Q&A]
### Type

Game behaviour

### Bug description

In seek-limited search boxes (such as the ones in the settings and song select), the cursor position cannot move normally, but when you hold control, it lets you move the cursor position by word. You can move the cursor normally in filter control text boxes (such as the one in the "now playing menu").

Also, the delete key does nothing in either search bar.

### Screenshots or videos

https://user-images.githubusercontent.com/55469508/180616686-a256973d-beff-4ce0-b97d-9c509b0a1fef.mp4

### Version

2022.723.0-lazer

### Logs

[performance-draw.log](https://github.com/ppy/osu/files/9174282/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/9174283/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/9174284/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/9174285/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9174286/updater.log)
[database.log](https://github.com/ppy/osu/files/9174287/database.log)
[network.log](https://github.com/ppy/osu/files/9174288/network.log)
[performance.log](https://github.com/ppy/osu/files/9174289/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/9174290/performance-audio.log)

**Comment by peppy** [comment]
@Squaduck what are your hopes / suggestions here? Would you want the ctrl-seeking to be disabled, or are you looking for more ability to seek instead?

**Reply by Squaduck** [reply]
I don't know. I just noticed that 1: seeking is disabled, and 2: there was a way to get around it. I'm not sure why seeking was disabled in the first place, so I don't know if it causes a bug. If it doesn't cause any issue, I would want more ability to seek.

**Reply by frenzibyte** [reply]
Seeking is disabled in places where the left/right is used for something else (i.e. when hovering a slider in settings, or when navigating through beatmaps in song select)

**Reply by peppy** [reply]
On checking, the code specifically allows use of left/right while holding ctrl and shift. I don't think it's such a bad thing, so I am not going to make a change for now. We'll see if others are confused by it, but the way it is now means you can perform text selection (and some degree of navigation) so it's a bit of a bonus.
